clang-tools-extra/clang-tidy/bugprone/BugproneTidyModule.cpp

```diff
@@ -139,6 +140,8 @@ class BugproneModule : public ClangTidyModule {
         "bugprone-inaccurate-erase");
     CheckFactories.registerCheck<IncorrectEnableIfCheck>(
         "bugprone-incorrect-enable-if");
+    CheckFactories.registerCheck<PublicEnableSharedFromThisCheck>(
+        "bugprone-public-enable-shared-from-this");
```

Comment:
name should be more a "non-public", as check should tell whats wrong.
i would call this check bugprone-incorrect-enable-shared-from-this, and then you could also check other constructions in future like:
```
class Bad2Example : enable_shared_from_this<BadExample> {};
```

Reply:
Or just call it bugprone-non-public-enable-shared-from-this

Reply:
Changed to bugprone-incorrect-enable-shared-from-this to keep the possibility of checking other constructions in the future open